azure/scope/machine.go

```diff
@@ -191,16 +194,21 @@ func (m *MachineScope) PublicIPSpecs() []azure.PublicIPSpec {
 }
 
 // InboundNatSpecs returns the inbound NAT specs.
-func (m *MachineScope) InboundNatSpecs() []azure.InboundNatSpec {
+func (m *MachineScope) InboundNatSpecs(existingRules []network.InboundNatRule) []azure.ResourceSpecGetter {
```

Comment:
let's add a comment to explain what existingRules means in this context and why it's here

Reply:
also wondering if we could avoid adding a dependency on the SDK in here by maybe passing in our own type that just has the rule name and frontend port (we would convert the []network. InboundNatRule in the service using a new func in the converters package). wdyt?